### ISSUE TITLE
FR: CLAP plugin support

### ISSUE THREAD

**Issue by dmt-hub** [mixxxdj/mixxx#13967]
### Feature Description

Support for loading CLAP plugins into MiXXX would be very nice. ( https://cleveraudio.org/1-feature-overview/ )
With the option to have (per loaded plugin) - 8 macro controls which show as knobs inside mixxx - which you can link to 8 knobs per plugin.
I noticed Audio Unit support will be implemented soon? Having 8 macro controls per plugin for AU plugins would be very interesting too.

**Comment by fwcd** [comment]
Our existing plugin backends (AU and LV2) use the effect system, so I think it would make most sense to implement any new plugin formats in the same way. Basically the plugin has to expose a set of parameters which can then be controlled in the usual manner via the UI or from a controller. Support for effect GUIs could also be added after we merge #13888.

**Comment by dmt-hub** [comment]
@fwcd Yes that would be perfect. Thank you!
Can we already experiment with Audio Units inside mixxx or is that function not ready yet?

**Comment by fwcd** [comment]
Yes AU support is in 2.5, but without GUIs.

**Comment by dmt-hub** [comment]
@fwcd 
I am using 2.5 beta and indeed i can load an AU, as there is no GUI yet it cannot be used atm? or is there a way to expose the plugin controls into the fx lane?

**Comment by fwcd** [comment]
Yes, you can control the plugin's parameters via the effect rack. GUI support is not merged yet.

**Comment by dmt-hub** [comment]
@fwcd How can i expose the parameters into the effect rack? i tried different AU plugins, and no parameters are shown.
<img width="713" alt="Screenshot 2024-12-04 at 12 13 51" src="https://github.com/user-attachments/assets/5cd9362c-773c-4514-a44e-5017678e3a45">

**Comment by fwcd** [comment]
Try the builds in #13945 and #13887

**Comment by dmt-hub** [comment]
@fwcd Sorry for my question, but can i download these builds somewhere or do i need to build it myself?

**Comment by dmt-hub** [comment]
Tried building from source before but always got stopped on this issue:
after doing cmake ..

-- Found WrapAtomic: TRUE
CMake Error at CMakeLists.txt:2174 (find_package):
  Could not find a package configuration file provided by "GTest" with any of
  the following names:

    GTestConfig.cmake
    gtest-config.cmake

  Add the installation prefix of "GTest" to CMAKE_PREFIX_PATH or set
  "GTest_DIR" to a directory containing one of the above files.  If "GTest"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!

 (trying to build on MacOS sonoma)

**Comment by fwcd** [comment]
You don't have to build from source, you can grab the GitHub Actions builds:

<img width="608" alt="Screenshot 2024-12-04 at 12 30 38" src="https://github.com/user-attachments/assets/8eef2804-9e1f-46b6-9744-91096a5a39e0">

<img width="592" alt="Screenshot 2024-12-04 at 12 31 36" src="https://github.com/user-attachments/assets/673a7036-a0df-47d7-907e-d9f9e4171c94">

W.r.t building from source, you should look into setting up a vcpkg buildenv, which contains all dependencies (e.g. gtest).

**Comment by dmt-hub** [comment]
@fwcd Thanks for your patience and help, downloading now!

**Comment by dmt-hub** [comment]
@fwcd Yes, thats working out fine. Noticed only the first 8 parameters are exposed, i guess thats to be expected at this stage of the implementation? 
Would be nice for future version of mixxx if there would be an option to choose which 8 parameters are exposed into the fx lane, so one can choose the most useful parameters to have in there.

**Comment by dmt-hub** [comment]
My request was linked to this issue, with macro knobs one could assign 8 favorite parameters from a plugin to each macro knob, these 8 macro knobs would be shown in the fx lane.

**Comment by fwcd** [comment]
That might be worth filing a separate issue for, since it could probably be implemented generically for all effects.
